client/views/account/accountProfile.coffee

```diff
@@ -63,7 +72,12 @@ Template.accountProfile.onCreated ->
 				reload = true
 
 			if _.trim $('#username').val()
-				data.username = _.trim $('#username').val()
+				if !RocketChat.settings.get("Accounts_AllowUsernameChange")
+					toastr.error t('Username_Change_Disabled')
+					instance.clearForm()
+					return
```

Comment:
This is the fix mentioned, now only triggers if a username has been entered rather than before where it would prevent password and language changes.